TRAMS-API/Controllers/ProjectsController.cs

```diff
@@ -82,27 +172,49 @@ public async Task<IActionResult> InsertTrust([FromBody]PostProjectsRequestModel
             {
                 foreach (var trustId in allTrustIds.Distinct())
                 {
-                    var trust = _trustsRepository.GetTrustById(trustId);
+                    var trustsRepositoryResult = await _trustsRepository.GetTrustById(trustId);
 
-                    if (trust == null)
+                    if (!trustsRepositoryResult.IsValid)
+                    {
+                        return _repositoryErrorHandler.LogAndCreateResponse(trustsRepositoryResult);
+                    }
+
+                    if (trustsRepositoryResult.Result == null)
                     {
                         unprocessableEntityErrors.Add($"No trust found with the id of: {trustId}");
                     }
                 }
             }
 
+            #endregion
+
+            //If errors are detected with entity referencing, return an UnprocessableEntity result
             if (unprocessableEntityErrors.Any())
             {
                 var error = string.Join(". ", unprocessableEntityErrors);
 
                 return UnprocessableEntity(error);
             }
 
-            var internalModel = _mapper.Map(model);
+            var internalModel = _postProjectsMapper.Map(model);
+
+            var insertProjectRepositoryResult = await _projectsRepository.InsertProject(internalModel);
+
+            if (!insertProjectRepositoryResult.IsValid)
+            {
+                return _repositoryErrorHandler.LogAndCreateResponse(insertProjectRepositoryResult);
+            }
+
+            var getProjectRepositoryResult = await _projectsRepository.GetProjectById(insertProjectRepositoryResult.Result.Value);
+
+            if (!getProjectRepositoryResult.IsValid)
+            {
+                return _repositoryErrorHandler.LogAndCreateResponse(getProjectRepositoryResult);
+            }
 
-            await _projectsRepository.InsertProject(internalModel);
+            var externalModel = _getProjectsMapper.Map(getProjectRepositoryResult.Result);
 
-            return NoContent();
+            return Created("entityLocation", externalModel);
```

Comment:
Question: Is this a placeholder for the actual location / url of the entity?

Reply:
Will fix